runtime/integration-tests/src/generic/utils/mod.rs

```diff
@@ -0,0 +1,136 @@
+use sp_runtime::traits::StaticLookup;
+pub mod genesis;
+
+use cfg_types::pools::TrancheMetadata;
+use frame_support::BoundedVec;
+use pallet_pool_system::tranches::{TrancheInput, TrancheType};
+use sp_runtime::{traits::One, Perquintill};
+
+use crate::generic::runtime::{Runtime, RuntimeKind};
+
+pub const POOL_MIN_EPOCH_TIME: Seconds = 24;
+
+pub fn give_nft<T: Runtime>(dest: AccountId, (collection_id, item_id): (CollectionId, ItemId)) {
+	pallet_uniques::Pallet::<T>::force_create(
+		RawOrigin::Root.into(),
+		collection_id,
+		T::Lookup::unlookup(dest.clone()),
+		true,
+	)
+	.unwrap();
+
+	pallet_uniques::Pallet::<T>::mint(
+		RawOrigin::Signed(dest.clone()).into(),
+		collection_id,
+		item_id,
+		T::Lookup::unlookup(dest),
+	)
+	.unwrap()
+}
+
+pub fn give_balance<T: Runtime>(dest: AccountId, amount: Balance) {
+	let data = pallet_balances::Account::<T>::get(dest.clone());
+	pallet_balances::Pallet::<T>::set_balance(
+		RawOrigin::Root.into(),
+		T::Lookup::unlookup(dest),
+		data.free + amount,
+		data.reserved,
+	)
+	.unwrap();
+}
+
+pub fn give_tokens<T: Runtime>(dest: AccountId, currency_id: CurrencyId, amount: Balance) {
+	let data = orml_tokens::Accounts::<T>::get(dest.clone(), currency_id);
+	orml_tokens::Pallet::<T>::set_balance(
+		RawOrigin::Root.into(),
+		T::Lookup::unlookup(dest),
+		currency_id,
+		data.free + amount,
+		data.reserved,
+	)
+	.unwrap();
+}
+
+pub fn give_pool_role<T: Runtime>(dest: AccountId, pool_id: PoolId, role: PoolRole) {
+	pallet_permissions::Pallet::<T>::add(
+		RawOrigin::Root.into(),
+		Role::PoolRole(role),
+		dest,
+		PermissionScope::Pool(pool_id),
+		Role::PoolRole(role),
+	)
+	.unwrap();
+}
+
+pub fn create_empty_pool<T: Runtime>(admin: AccountId, pool_id: PoolId, currency_id: CurrencyId) {
+	pallet_pool_registry::Pallet::<T>::register(
+		match T::KIND {
+			RuntimeKind::Development => RawOrigin::Signed(admin.clone()).into(),
+			_ => RawOrigin::Root.into(),
+		},
+		admin,
+		pool_id,
+		vec![
+			TrancheInput {
+				tranche_type: TrancheType::Residual,
+				seniority: None,
+				metadata: TrancheMetadata {
+					token_name: BoundedVec::default(),
+					token_symbol: BoundedVec::default(),
+				},
+			},
+			TrancheInput {
+				tranche_type: TrancheType::NonResidual {
+					interest_rate_per_sec: One::one(),
+					min_risk_buffer: Perquintill::from_percent(0),
+				},
+				seniority: None,
+				metadata: TrancheMetadata {
+					token_name: BoundedVec::default(),
+					token_symbol: BoundedVec::default(),
+				},
+			},
+		],
+		currency_id,
+		Balance::MAX,
+		None,
+		BoundedVec::default(),
+	)
+	.unwrap();
+
+	// In order to later close the epoch fastly,
+	// we mofify here that requirement to significalty reduce the testing time.
+	// The only way to do it is breaking the integration tests rules mutating
+	// this state directly.
+	pallet_pool_system::Pool::<T>::mutate(pool_id, |pool| {
+		pool.as_mut().unwrap().parameters.min_epoch_time = POOL_MIN_EPOCH_TIME;
+	});
```

Comment:
@mustermeiszer, my question about epoch time in slack came from this. If I call:
```rust
env.pass(Blocks::BySeconds(T::DefaultMinEpochTime::get()))
```
it takes around 10 min per test, even without fudge, only passing blocks to reach the point in time where I can actually closing the epoch.

If I modify the pool to set to the min bound, it takes around 60 seconds, but I think it's still not feasible if we expect to have a lot of loans tests (or any other thing that requires a funded pool).

The solution was to hardcode a really low value, but I don't like it because I should use something more high-level to deal with it. Doing this is really obscure.